examples/pennylane_run_qaoa_maxcut.py

```diff
@@ -0,0 +1,287 @@
+        U_B(params[1][i]) # params[1] is beta
+    if edge is None:
+        # measurement phase
+        return qml.sample(comp_basis_measurement(range(n_wires)))
+    # during the optimization phase we are evaluating a term
+    # in the objective using exp val
+    return qml.expval(qml.Hermitian(pauli_z_2, wires=edge))
+
+
+##############################################################################
+# Optimization
+# ~~~~~~~~~~~~
+# Finally, we optimize the objective over the
+# angle parameters :math:`\boldsymbol{\gamma}` (``params[0]``) and :math:`\boldsymbol{\beta}`
+# (``params[1]``).
+# and then sample the optimized
+# circuit multiple times to yield a distribution of bitstrings. One of the optimal partitions
+# (:math:`z=0101` or :math:`z=1010`) should be the most frequently sampled bitstring.
+
+
+def qaoa_maxcut(n_layers=1):
+    print("\np={:d}".format(n_layers))
+
+    # initialize the parameters near zero
+    init_params = 0.01 * np.random.rand(2, n_layers)
+
+    # minimize negative the objective
+    def objective(params):
+        neg_obj = 0
+        for edge in graph:
+            # objective for the maxcut problem
+            neg_obj -= 0.5 * (1 - circuit(params, edge=edge, n_layers=n_layers))
+        return neg_obj
+
+    # initialize optimizer
+    opt = qml.AdagradOptimizer(stepsize=0.5)
+
+    # optimize parameters in objective
+    params = init_params
+    steps = 30
+    for i in range(steps):
+        params = opt.step(objective, params)
+        if (i + 1) % 5 == 0:
+            print("Objective after step {:5d}: {: .7f}".format(i + 1, -objective(params)))
+
+    # sample measured bitstrings 100 times
+    bit_strings = []
+    n_samples = 100
+    for i in range(0, n_samples):
+        bit_strings.append(int(circuit(params, edge=None, n_layers=n_layers)))
+
+    # print optimal parameters and most frequently sampled bitstring
+    counts = np.bincount(np.array(bit_strings))
+    most_freq_bit_string = np.argmax(counts)
+    print("Optimized (gamma,beta) vectors: {}".format(params))
+    print("Most frequently sampled bit string is: {:04b}".format(most_freq_bit_string))
+
+    return -objective(params), bit_strings
+
+
+# perform qaoa on our graph with p=1,2 and
+# keep the bitstring sample lists
+bitstrings1 = qaoa_maxcut(n_layers=1)[1]
+bitstrings2 = qaoa_maxcut(n_layers=2)[1]
+
+##############################################################################
+# In the case where we set ``n_layers=2``, we recover the optimal
+# objective function :math:`C=4`
+
+##############################################################################
+# Plotting the results
+# --------------------
+# We can plot the distribution of measurements we got from the optimized circuits. As
+# expected for this graph, the partitions 0101 and 1010 are measured with the highest frequencies,
+# and in the case where we set ``n_layers=2`` we obtain one of the optimal partitions with 100% certainty.
```

Comment:
I want to understand: how come we get the right answers when the computational basis measurement is not working properly in PL? Or am I missing something?

Reply:
the comp_basis_measurement function defined above is the ad hoc fix